### PR TITLE
Update GHA

### PR DESCRIPTION
**Pull request by ocefpaf**
**What does your PR do?**

Update and modernize the GitHub Actions here.
I'll add pre-commits in a bit, just testing if these are passing first.

**What issue does this PR address**

Test suites are not passing and are a nit outdated.

**Testing**

No new functionality but I remove the flake tests in lieu of pre-commits.

Closes https://github.com/callumrollo/cmcrameri/pull/9

### CODE REVIEW

**Comment by ocefpaf** [comment]
@callumrollo you can enable pre-commit-ci in https://pre-commit.ci/ and never worry about code formatting ever again ;-p

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Publish to PyPI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (ocefpaf) [inline .github/workflows/pypi.yml:7]: @callumrollo let me know if you are OK renaming master to main. If not I can change this.

**Comment by callumrollo** [comment]
main is now the default branch

**Comment by ocefpaf** [comment]
Tests are online again. I rebased this one against #15 and they should pass now. Fingers crossed :crossed_fingers:

**Pull request merged**